src/components/Portfolio/portfolioItem.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react'
+import { Link, useHistory } from 'react-router-dom'
+import { useLocation } from 'react-router'
+import { updatePortfolioData } from '../../redux/actions'
+import { useDispatch, useSelector } from 'react-redux'
+import { removeFromPortfolio } from '../../api/portfolio'
+import { currencySymbolMap } from '../../assets/constants'
+import GraphDown from '../../assets/images/downwards-light-desktop.svg'
+import GraphUp from '../../assets/images/upwards-light-desktop.svg'
+import GraphStraight from '../../assets/images/straight-light-desktop.svg'
+import Footer from '../Other/footer'
+import { updateItemInfo, setItemPricesLoading, setItemListingsLoading } from '../../redux/actions'
+
+export default function PortfolioItem() {
+
+    const dispatch = useDispatch()
+    const location = useLocation()
+    const history = useHistory()
+
+    const portfolio = useSelector(state => state.portfolio)
+    const currency = useSelector(state => state.currency)
+
+    const item = location.itemInfo
+    const modelName = item.data.modelName
+    const recordID = item.record_id
+    const image = item.data.image
+    const price = Math.round(item.price)
+    const currentPrice = Math.round(item.currentPrice)
+    const priceChange = currentPrice - price
+    const percentChange = (priceChange / price * 100).toFixed(2)
+    const colour = priceChange === 0 ? 'grey' : (priceChange > 0 ? 'green' : 'red')
+    const graph = priceChange === 0 ? GraphStraight : (priceChange > 0 ? GraphUp : GraphDown)
+
+    // TODO: add code to link to item page for this sneaker
+    const generateLink = () => {
+        dispatch(updateItemInfo({}))
+        dispatch(setItemPricesLoading(true))
+        dispatch(setItemListingsLoading(true))
+        
+        let itemKey = item.data.sku ? encodeURIComponent(item.data.sku) : item.data.urlKey 
+        return item.gender === null ? `/item/${itemKey}` : `/item/${itemKey}/${item.gender}`
+    }
+
+    const getNavData = () => {
+        let itemInfo = {
+            sku: item.sku ? item.sku.replaceAll('-', ' ') : null,
+            modelName: item.data.modelName,
+            image: item.data.image,
+            url: item.data.url,
+            urlKey: item.data.urlKey
+        }
+        return itemInfo
+    }
+
+    const removeItemHandler = (recordID) => {
```

Comment:
This function is also defined in `portfolio.js`, could you abstract it out to one place so it's not defined in two different locations?